### PR TITLE
Finalise test remake

### PR DESCRIPTION
**Pull request by TorkelE**
Continuation of https://github.com/SciML/Catalyst.jl/pull/622

### CODE REVIEW

File: test/reactionsystem_structure/reactionsystem.jl [hunk 1]
```diff
@@ -137,25 +137,28 @@ end
...
-G2 = sf.g(u, p, t)
-@test norm(G - G2) < 100 * eps()
-
-# Test conversion to NonlinearSystem.
-ns = convert(NonlinearSystem, rs)
-fnl = eval(generate_function(ns)[2])
-dunl = similar(du)
-fnl(dunl, u, p)
-@test norm(du - dunl) < 100 * eps()
+let
```
Comment (isaacsas) [inline test/reactionsystem_structure/reactionsystem.jl:142]: What about if you do 
```julia
let t = 0.0
```

Comment (TorkelE) [reply test/reactionsystem_structure/reactionsystem.jl:142]: That might be worth trying, will try tomorrow morning.

Comment (TorkelE) [reply test/reactionsystem_structure/reactionsystem.jl:142]: Ok, unfortunately this didn't work.

Comment (isaacsas) [reply test/reactionsystem_structure/reactionsystem.jl:142]: What's the error?

Comment (TorkelE) [reply test/reactionsystem_structure/reactionsystem.jl]: These two blocks could be put in a "let .. end" statement, however, this causes "fnl(dunl, u, p)" to throw a "MethodError: no method matching Float64(::Num)" error.

**Pull request merged**